velox/exec/RowContainer.h

```diff
@@ -640,45 +715,133 @@ class RowContainer {
     }
   }
 
-  template <typename T>
+  template <bool useRowNumbers, typename T>
   static void extractValuesWithNulls(
       const char* FOLLY_NONNULL const* FOLLY_NONNULL rows,
+      folly::Range<const vector_size_t*> rowNumbers,
       int32_t numRows,
       int32_t offset,
       int32_t nullByte,
       uint8_t nullMask,
+      int32_t resultOffset,
       FlatVector<T>* FOLLY_NONNULL result) {
-    result->resize(numRows);
-    BufferPtr nullBuffer = result->mutableNulls(numRows);
+    int32_t numRowsInternal = numRows;
+    if constexpr (useRowNumbers) {
+      numRowsInternal = rowNumbers.size();
+    }
+    auto maxRows = numRowsInternal + resultOffset;
+    result->resize(maxRows);
+    BufferPtr nullBuffer = result->mutableNulls(maxRows);
     auto nulls = nullBuffer->asMutable<uint64_t>();
-    BufferPtr valuesBuffer = result->mutableValues(numRows);
+    BufferPtr valuesBuffer = result->mutableValues(maxRows);
     auto values = valuesBuffer->asMutableRange<T>();
-    for (int32_t i = 0; i < numRows; ++i) {
-      if (rows[i] == nullptr) {
-        bits::setNull(nulls, i, true);
+    for (int32_t i = 0, j = resultOffset; i < numRowsInternal; ++i, ++j) {
+      const char* row;
+      if constexpr (useRowNumbers) {
+        row = rows[rowNumbers[i]];
+      } else {
+        row = rows[i];
+      }
+      if (row == nullptr) {
+        bits::setNull(nulls, j, true);
       } else {
-        bits::setNull(nulls, i, isNullAt(rows[i], nullByte, nullMask));
-        values[i] = valueAt<T>(rows[i], offset);
+        bits::setNull(nulls, j, isNullAt(row, nullByte, nullMask));
+        values[j] = valueAt<T>(row, offset);
       }
     }
   }
 
-  template <typename T>
+  template <bool useRowNumbers>
+  static void extractValuesWithNullsStringView(
```

Comment:
Can we avoid having this method by adding if constexpr (std::is_same_v<T, StringView>) branch to the generic template?

Reply:
I had styled based on previous code. But this usage does reduce code-duplication quite a bit.